src/raftstore/store/peer.rs

```diff
@@ -824,6 +824,8 @@ impl Peer {
                         "{} becomes leader and lease expired time is {:?}",
                         self.tag, self.leader_lease
                     );
+                    // make approximate size and keys updated after leader transfer, cause new leader may have a stale size and keys.
+                    self.size_diff_hint = self.cfg.region_split_check_diff.0;
```

Comment:
This only triggers the region size evaluation, but the evaluation is done in a single thread. Will it not be timely enough when a large number of leader switches?

Reply:
Carry the region size in the heartbeat of the raft, cache the region size in the copy, and get the correct size immediately when the leader transfers

Reply:
yeah, it is still probable that tikv reports a stale size to PD. But it is acceptable and won't last too much time comparing with current implementation.

Reply:
How many region transfer have you tested at most, we now have a cluster of single node with a number of regions of 100,000

Reply:
@Connor1996 How about calculate approximate size not only for leader, but also for follower? https://github.com/tikv/tikv/blob/master/src/raftstore/store/fsm/peer.rs#L1715 Better to test if this can produce any bad effects.

Reply:
@zhongleihe Thanks for advice, I will implement it in another way later.